src/fondant/component/executor.py

```diff
@@ -75,6 +78,15 @@ def __init__(
         self.metadata = Metadata.from_dict(metadata)
         self.user_arguments = user_arguments
         self.input_partition_rows = input_partition_rows
+        self.schema = schema
+
+        # TODO: either overwrite the component spec
```

Comment:
Looking for some input on this todo. Either we are overwrite the component spec, or we passing the consumes and produces further to data_io. Where could use the attribute values of consumes and produces if available instead of the component spec. 
Not sure which one is the better approach.